### PR TITLE
ignore_errors: Use template variable values

### LINKED ISSUE #56345: `ignore_errors: {{ ansible_check_mode }}` is always true
##### SUMMARY
When having a task which uses `ignore_errors: {{ ansible_check_mode }}` and the task results in an exception the error is always ignored. 
This is especially bad when running in non-check mode.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
task_executor

##### ANSIBLE VERSION

```
ansible 2.7.10
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/cedric/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/cedric/test/venv/lib/python2.7/site-packages/ansible
  executable location = /home/cedric/test/venv/bin/ansible
  python version = 2.7.5 (default, Apr 11 2018, 07:36:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
```

##### CONFIGURATION

```paste below
nothing
```

##### STEPS TO REPRODUCE

```yaml
- name: bug on ignore_errors and ansible_check_mode
  hosts: localhost
  gather_facts: false

  tasks:
    - debug:
        var: ansible_check_mode

    - debug:
        msg: "foo = {{ foo }}"
      when: foo
      ignore_errors: "{{ ansible_check_mode }}"
```

##### EXPECTED RESULTS

Ignore errors only in the check mode, as indicated in the documentation https://docs.ansible.com/ansible/latest/user_guide/playbooks_checkmode.html?information-about-check-mode-in-variables#information-about-check-mode-in-variables

##### ACTUAL RESULTS

```paste below
(venv) cedric@jacksbastion ~/test $ ansible-playbook test.yml -vv
ansible-playbook 2.7.10
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/cedric/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/cedric/test/venv/lib/python2.7/site-packages/ansible
  executable location = /home/cedric/test/venv/bin/ansible-playbook
  python version = 2.7.5 (default, Apr 11 2018, 07:36:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
Using /etc/ansible/ansible.cfg as config file
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


PLAYBOOK: test.yml *****************************************************************************************************************************************************************************************
1 plays in test.yml

PLAY [bug on ignore_errors and ansible_check_mode] *********************************************************************************************************************************************************
META: ran handlers

TASK [debug] ***********************************************************************************************************************************************************************************************
task path: /home/cedric/test/test.yml:6
ok: [localhost] => {
    "ansible_check_mode": false
}

TASK [debug] ***********************************************************************************************************************************************************************************************
task path: /home/cedric/test/test.yml:9
fatal: [localhost]: FAILED! => {"msg": "The conditional check 'foo' failed. The error was: error while evaluating conditional (foo): 'foo' is undefined\n\nThe error appears to have been in '/home/cedric/test/test.yml': line 9, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - debug:\n      ^ here\n"}
...ignoring
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0
```
The `foo` variable is undefined but the play ignores the error `...ignoring`

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Templatize the variable before using them in ignore_errors directive.

Fixes: #56345

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/56345-ignore_errors.yml
lib/ansible/plugins/strategy/__init__.py
test/integration/targets/ignore_errors/tasks/main.yml

### CODE REVIEW

**Comment by Akasurde** [comment]
@sivel Could you please review this? Thanks in advance.

**Review by jamescassell — APPROVED** [review]
shipit

**Comment by Akasurde** [comment]
This is not the correct fix. Will open a new PR with correct fix.

File: lib/ansible/plugins/strategy/__init__.py [hunk 1]
```diff
@@ -487,6 +487,14 @@ def search_handler_blocks_by_name(handler_name, handler_blocks):
             if task_result.is_failed():
                 role_ran = True
                 ignore_errors = original_task.ignore_errors
+                if handler_templar.is_template(ignore_errors):
```
Comment (bcoca) [inline lib/ansible/plugins/strategy/__init__.py:490]: just modify the previous line to post_validate ignore_errors

**Pull request closed without merging**